app/routers/api/files.py

```
@@ -128,7 +128,137 @@ def _serialize_file(
     return file_dict
 
 
-@router.get("")
+def _parse_tag_ids(tag_ids: Optional[str]) -> Optional[List[int]]:
+    """Parse comma-separated tag IDs string into a list of integers."""
+    if not tag_ids:
+        return None
+    try:
+        return [int(tid.strip()) for tid in tag_ids.split(",") if tid.strip()]
+    except ValueError:
+        raise ValueError("Invalid tag_ids format. Must be comma-separated integers.")
```

Ruff (0.15.9)

[warning] Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)